pymeasure/instruments/validators.py

```diff
@@ -40,6 +40,30 @@ def strict_range(value, values):
         ))
 
 
+def strict_discrete_range(value, values, step):
```

Comment:
Please add tests in [test_validators.py](https://github.com/ralph-group/pymeasure/blob/master/tests/instruments/test_validators.py) to cover all use cases and edge cases.